### ISSUE TITLE
Rate limiter

### ISSUE THREAD

**Issue by Beholder1984** [Schlabbi/docker-notify#19]
I am trying to monitor 9 docker containers with docker-notify. After adding the last project docker-notify started spamming me with error messages. In the result from docker-hub I see messages containing the following:
```
statusCode: 429,
statusMessage: 'Too Many Requests',
```
Which suggests that I am making to many requests to Docker hub. Any suggestions on how I can monitor more then 8 projects?

**Comment by Schlabbi** [comment]
That's a bug. I should rework the DockerAPI implementation. Maybe we can also reduce the amount of requests that are currently made so we don't hit the rate limit as easily. So sorry, but until that is fixed, you will probably run into these errors.

**Comment by rgstephens** [comment]
@Beholder1984 What is your `checkInterval`?  I'm monitoring 9 images with a 20 minute interval and haven't run into this problem.

**Comment by Beholder1984** [comment]
@rgstephens my checkInterval is currently set to 60. Do you by any change have a subscription to docker hub. It seems that paying customers won't hit the rate limiter so fast.

**Comment by rgstephens** [comment]
No, I don't have a paid subscription and the code is not logging in with an id either.

**Comment by TonyRL** [comment]
https://github.com/TonyRL/docker-notify/pull/33 adds support for optional Docker Hub auth.
You can now use your own personal plan account for double the limits (200 checks/6hrs vs 100 checks/6hrs).
FYI, You can also use githubactions' account as it does not any rate limit.
